frontend/src/modules/Metadata_Forms/components/MetadataFormFields.js

```diff
@@ -0,0 +1,424 @@
+              sx={{ mt: 1 }}
+              onClick={startEdit}
+              type="button"
+              variant="outlined"
+            >
+              Edit
+            </Button>
+          </TableCell>
+        </TableRow>
+      </TableHead>
+      <TableBody>
+        {fields.length === 0 ? (
+          <TableRow>
+            <TableCell colSpan={6} align="center">
+              No fields found
+            </TableCell>
+          </TableRow>
+        ) : (
+          fields.map((field) => (
+            <TableRow key={field.uri}>
+              <TableCell
+                sx={{
+                  textAlign: 'center'
+                }}
+              >
+                {field.required ? <AsteriskIcon /> : ''}
+              </TableCell>
+              <TableCell>{field.name}</TableCell>
+              <TableCell>{field.type}</TableCell>
+              <TableCell>{field.description}</TableCell>
+              <TableCell>{field.possibleValues}</TableCell>
+              <TableCell></TableCell>
+            </TableRow>
+          ))
+        )}
+      </TableBody>
+    </Table>
+  );
+};
+DisplayTable.propTypes = {
+  fields: PropTypes.array.isRequired,
+  startEdit: PropTypes.func.isRequired
+};
+
+export const MetadataFormFields = (props) => {
+  const dispatch = useDispatch();
+  const client = useClient();
+  const { metadataForm, fieldTypeOptions } = props;
+  const [loading, setLoading] = useState(false);
+  const [editOn, setEditOn] = useState(false);
+  const [fields, setFields] = useState(metadataForm.fields);
+  const [inputValue, setInputValue] = useState('');
+  const [filter, setFilter] = useState({});
+
+  const handleInputChange = (event) => {
+    setInputValue(event.target.value);
+    setFilter({ ...filter, term: event.target.value });
+  };
+
+  const startEdit = () => {
+    setEditOn(true);
+  };
+
+  const fetchItems = async () => {
+    setLoading(true);
+    const response = await client.query(getMetadataForm(metadataForm.uri));
+    if (!response.errors && response.data.getMetadataForm !== null) {
+      setFields(response.data.getMetadataForm.fields);
+    } else {
+      const error = response.errors
+        ? response.errors[0].message
+        : 'Metadata Forms not found';
+      dispatch({ type: SET_ERROR, error });
+    }
+    setLoading(false);
+  };
+
+  const saveChanges = async (updatedFields) => {
+    setLoading(true);
+    // remove new fields (not yet saved in DB), that were deleted during editing
+    const data = updatedFields.filter((field) => !field.deleted || field.uri);
+    data.forEach((field) => {
+      delete field.__typename;
+    });
+    const response = await client.mutate(
+      batchMetadataFormFieldUpdates(metadataForm.uri, data)
+    );
+    if (
+      !response.errors &&
+      response.data &&
+      response.data.batchMetadataFormFieldUpdates !== null
+    ) {
+      setFields(response.data.batchMetadataFormFieldUpdates);
```

Comment:
this setFields works for immediate changes reflected after an update to the fields for the form

But if I then click away to a new tab (say `Form Info`) and then click back to `Fields` I see the old fields pre-update (and will remain that way until I refresh the page

Believe this is because the next time we open `Fields` tab without refresh between form field edits --> we are using the resolved fields from the original (un-edited) form (i.e. `const [fields, setFields] = useState(metadataForm.fields);`) and these are not updated by the response of `batchMetadataFormFieldUpdates`

Reply:
Good catch

Reply:
I added useEffect [340aea0](https://github.com/data-dot-all/dataall/pull/1455/commits/340aea0e7d9d67c1144e0e35f470b3a5a9deed02)